bpf/init.sh

```diff
@@ -140,19 +125,15 @@ function move_local_rules()
 
 function setup_proxy_rules()
 {
-	# delete old ip rules and flush table
-	delete_old_ip_rules $PROXY_RT_TABLE
+	# Any packet from a local process uses a separate routing table
+	rulespec="fwmark 0xA00/0xF00 pref 10 lookup $PROXY_RT_TABLE"
 
-	if [ -n "$(ip -4 rule list)" ]; then
-		ip -4 route flush table $PROXY_RT_TABLE
-		# Any packet from a proxy uses a separate routing table
-		ip -4 rule add fwmark 0xA00/0xF00 pref 10 lookup $PROXY_RT_TABLE
+	if [ -z "$(ip -4 rule list $rulespec)" ]; then
+		ip -4 rule add $rulespec
```

Comment:
Do these need to be `|| true` to handle `EEXIST`?

```
$ sudo ip rule add fwmark 0xA00/0xF00 pref 10 lookup 2005
$ echo $?
0
$ sudo ip rule add fwmark 0xA00/0xF00 pref 10 lookup 2005
RTNETLINK answers: File exists
$ echo $?
2
```

Reply:
That's why it checks if the rule already exists with `ip rule list`. Unfortunately there is no `ip rule replace`.